### PR TITLE
Template optimization

### PR DESCRIPTION
**Pull request by joshua-stone**
This change brings some optimizations to total ISO build times in a couple ways:

A) switch from the fedora container to the fedora-minimal image which will download faster and has the dnf5 command which runs significantly faster than dnf
B) switch from ansible to jinja2 which has fewer dependencies and runs faster

### CODE REVIEW

File: isopatch.sh [hunk 1]
```diff
@@ -65,44 +65,57 @@ patch_grub_cfg() {
...
-                dest=${EFI_ROOT}/${GRUB_CFG}" \
-            localhost
+        sudo jinja2 --strict \
+            -o "${EFI_ROOT}/${GRUB_CFG}" \
+            -D "BOOT_TYPE=efi" \
+            -D "VOLUME_ID=${VOLUME_ID}" \
+            -D "RELEASE=${RELEASE}" \
+            "${SCRIPT_DIR}/installer/overlay/${ARCH}/${GRUB_CFG}" \
+            boot_menu.yml
+        #ansible -m template \
```
Comment (akdev1l) [inline isopatch.sh]: Can we remove these leftovers?

File: installer/overlay/x86_64/EFI/BOOT/grub.cfg [hunk 2]
```diff
@@ -58,21 +56,21 @@ submenu 'Install {{ submenu.label }} -->' {
...
 }
-{% endfor %}
+{%- endfor %}
 submenu 'Troubleshooting -->' {
 	menuentry 'Rescue a ublue-os system' --class fedora --class gnu-linux --class gnu --class os {
 		linux{% if BOOT_TYPE == 'efi' %}efi{% endif %} /images/pxeboot/vmlinuz inst.stage2=hd:LABEL={{ VOLUME_ID }} inst.rescue quiet
 		initrd{% if BOOT_TYPE == 'efi' %}efi{% endif %} /images/pxeboot/initrd.img
 	}
-{% if BOOT_TYPE == 'mbr' %}
+{%- if BOOT_TYPE == 'mbr' %}
```
Comment (akdev1l) [inline installer/overlay/x86_64/EFI/BOOT/grub.cfg:68]: What does this do?🤔

File: action.yml [hunk 3]
```diff
@@ -22,16 +22,17 @@ runs:
...
       shell: bash
       run: |
-        dnf install \
+        dnf5 install \
           --disablerepo='*' \
           --enablerepo='fedora,updates' \
           --setopt install_weak_deps=0 \
           --assumeyes \
-          ansible \
           curl \
```
Comment (akdev1l) [inline action.yml]: not really related but I wonder if we can reuse the deps.txt here so  we just have 1 dependency list

**Review by akdev1l — COMMENTED** [review]
I think we should remove the ansible commands, otherwise looks good!

File: installer/overlay/x86_64/EFI/BOOT/grub.cfg (hunk 2 above)
Comment (joshua-stone) [reply installer/overlay/x86_64/EFI/BOOT/grub.cfg:68]: This is part of jinja2's whitespace control:

https://jinja.palletsprojects.com/en/latest/templates/#whitespace-control

File: action.yml (hunk 3 above)
Comment (joshua-stone) [reply action.yml]: It should be downloading dependencies in a similar fashion as the [`shellcheck`](https://github.com/ublue-os/isogenerator/blob/main/.github/workflows/shellcheck.yml) job now.

File: installer/overlay/x86_64/EFI/BOOT/grub.cfg (hunk 2 above)
Comment (akdev1l) [reply installer/overlay/x86_64/EFI/BOOT/grub.cfg:68]: that is awesome

**Review by akdev1l — APPROVED** [review]

**Pull request merged**